cluster-autoscaler/simulator/drainability/rules/rules.go

```diff
@@ -86,10 +86,23 @@ func (rs Rules) Drainable(drainCtx *drainability.DrainContext, pod *apiv1.Pod) d
 		drainCtx.RemainingPdbTracker = pdb.NewBasicRemainingPdbTracker()
 	}
 
+	var candidates []drainability.Status
+
 	for _, r := range rs {
-		d := r.Drainable(drainCtx, pod)
-		if d.Outcome != drainability.UndefinedOutcome {
-			return d
+		status := r.Drainable(drainCtx, pod)
+		if len(status.Overrides) > 0 {
+			candidates = append(candidates, status)
+			continue
+		}
+		for _, candidate := range candidates {
+			for _, override := range candidate.Overrides {
+				if status.Outcome == override {
```

Comment:
Maybe worth logging the fact that override happened? At least at `V(5)`? This logic becomes non-trivial with this change, so some observability wouldn't hurt.

Reply:
Great idea. Added a Name() to the Rule interface to make this log more useful.